modules/tinymce/src/core/test/ts/browser/EditorAutoFocusTest.ts

```diff
@@ -0,0 +1,63 @@
+import { describe, it, before, afterEach } from '@ephox/bedrock-client';
+import { Insert, SugarBody, SugarElement } from '@ephox/sugar';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+import { EditorManager } from 'tinymce/core/api/PublicApi';
+
+const isInViewport = (editor: Editor) => {
+  const containerRect = editor.getContentAreaContainer().getBoundingClientRect();
+  const innerRect = editor.selection.getNode().getBoundingClientRect();
+  const top = containerRect.top + innerRect.top + 1;
+  return top > 0 && top + innerRect.height < window.innerHeight;
+};
+
+const setupEditorAutoFocus = (id: string) => {
```

Comment:
Per the naming conversions, anything that's async in tests should be prefixed with `p`. E.g. `pSetupEditorAutoFocus`.